tests/test_text.py

```diff
@@ -0,0 +1,464 @@
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        # "diverse" is Default cost metric
+        method = SelectionMethod.TextBased(num_features=None,
+                                           featurization_method=TextWiser(Embedding.TfIdf(min_df=0),
+                                                                          [Transformation.NMF(n_components=20),
+                                                                           Transformation.SVD(n_components=10)]),
+                                           optimization_method="random",
+                                           trials=5)
+
+        selector = Selective(method)
+        selector.fit(data, labels)
+        selected_features = selector.transform(data)
+
+        self.assertTrue(isinstance(selected_features, pd.DataFrame))
+        self.assertEqual(selected_features.shape[0], data.shape[0])
+        if method.num_features is None:
+            pass
+        else:
+            self.assertEqual(selected_features.shape[1], 2)
+
+    ################################################
+    ########## Tests for greedy selection ##########
+    ################################################
+    def test_text_based_greedy_num_feature_unicost(self):
+        data = pd.DataFrame(
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [1, 0, 0, 0, 1], "item2": [0, 1, 0, 1, 0], "item3": [1, 0, 1, 0, 0],
+                               "item4": [1, 0, 0, 0, 0], "item5": [0, 1, 0, 0, 1]})
+
+        method = SelectionMethod.TextBased(num_features=2,
+                                           optimization_method="greedy",
+                                           cost_metric="unicost")
+
+        selector = Selective(method)
+        selector.fit(data, labels)
+        selected_features = selector.transform(data)
+
+        assert selector.selection_method.trials == 10
+        self.assertTrue(isinstance(selected_features, pd.DataFrame))
+        self.assertEqual(selected_features.shape[0], data.shape[0])
+        if method.num_features is None:
+            pass
+        else:
+            self.assertEqual(selected_features.shape[1], 2)
```

Comment:
We need to verify the selected items in these, no? Not just the size? Cannot you setup the labels columns such that we know the greedy solution clearly, and then assert the selections here. 

Immediate exampels would be, a greedy method with Idenitiy labels -- it should select all. 
Another is greedy with a single coluimn covering everying --it should definitely select that index 

And what happens with greedy when 1) there is no label coverage at all 2) some labels cannot just be covered (no item has it, the instance is infeasible)

Reply:
Please check the tests. The mentioned tests are added for the tests based on their relevance.